### PR TITLE
fix(view): isLoaded stability improvements

### PR DESCRIPTION
**Pull request by NathanWalker**
## PR Checklist

- [x] The PR title follows our guidelines: https://github.com/NativeScript/NativeScript/blob/master/CONTRIBUTING.md#commit-messages.
- [x] There is an issue for the bug/feature this PR is for. To avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it.
- [x] You have signed the [CLA](http://www.nativescript.org/cla).
- [x] All existing tests are passing: https://github.com/NativeScript/NativeScript/blob/master/DevelopmentWorkflow.md#running-unit-tests.
- [ ] Tests for the changes are included - https://github.com/NativeScript/NativeScript/blob/master/WritingUnitTests.md.

## What is the current behavior?

App's can randomly crash when various bindings update upon resuming an app dependent upon a wide array of possible view setups via developers.

## What is the new behavior?

Guards have been added to account for various troubling race conditions.

closes https://github.com/NativeScript/NativeScript/issues/6179

### CODE REVIEW

**Comment by MartoYankov** [comment]
test

File: tns-core-modules/ui/frame/frame.android.ts [hunk 1]
```diff
@@ -718,7 +718,7 @@ class FragmentCallbacksImplementation implements AndroidFragmentCallbacks {
             this.frame._addView(page);
         }
 
-        if (frame.isLoaded && !page.isLoaded) {
+        if ((frame && frame.isLoaded) && (page && !page.isLoaded)) {
```
Comment (vakrilov) [inline tns-core-modules/ui/frame/frame.android.ts]: `page` is used couple of lines above - so no point of guarding it here. If it was `undefined` the code would have crashed [here](https://github.com/NativeScript/NativeScript/pull/6187/files#diff-c1ad77120477acf9bacdaabe6af2518dR705)

File: tns-core-modules/ui/styling/style-scope.ts [hunk 2]
```diff
@@ -357,7 +357,7 @@ export class CssState {
      * As a result, at some point in time, the selectors matched have to be requerried from the style scope and applied to the view.
      */
     public onChange(): void {
-        if (this.view.isLoaded) {
+        if (this.view && this.view.isLoaded) {
```
Comment (vakrilov) [inline tns-core-modules/ui/styling/style-scope.ts]: What are the cases when you get error here?
The `view` is passed in the constructor and never changed. 
Maybe we should add a check there?

File: tns-core-modules/ui/frame/frame.android.ts (hunk 1 above)
Comment (NathanaelA) [reply tns-core-modules/ui/frame/frame.android.ts]: I suspect that because frame can be null here; that he just safe guarded both at the same time.   Probably should also have the "page" test on the lines above it too, in case that is ever null.

File: tns-core-modules/ui/styling/style-scope.ts (hunk 2 above)
Comment (NathanaelA) [reply tns-core-modules/ui/styling/style-scope.ts]: Well, onChange is called because of css changes; if css is in the process of changing and the view is being de-allocated it will crash because this.view = null.

Comment (NathanWalker) [reply tns-core-modules/ui/styling/style-scope.ts]: That’s exactly right @nathanaela

File: tns-core-modules/ui/frame/frame.android.ts (hunk 1 above)
Comment (NathanWalker) [reply tns-core-modules/ui/frame/frame.android.ts]: @vakrilov @NathanaelA I updated the handling to ensure of number of possible errors that could occur due to timing and null/undefined - also added trace handling to ensure the log is notified in event of such conditions.

**Comment by NathanWalker** [comment]
Worth noting I am using these changes in a custom pack of core modules in a current project to solve the issues reported and is working 100% without error with these in place.

**Comment by MartoYankov** [comment]
We should consider merging this in the `release` branch.

File: tns-core-modules/ui/styling/style-scope.ts (hunk 2 above)
Comment (vakrilov) [reply tns-core-modules/ui/styling/style-scope.ts]: I'm still struggling to get this one. The `view` is private and is never set (except from the constructor).
If there is a problem with trying to apply styles on disposed view - this check will do nothing more. The view will be disposed, but its `CssState` will still hold the reference and so `this.view` will not be null.

**Comment by vakrilov** [comment]
Hey @NathanWalker and @NathanaelA. I will cherry pick the commit and open a PR in release, so that we can include it in a patch release. 

There are still some changes that don't really make sense to me and I'm trying to understand if you have crash reports related to them (the stack trace form the issue points only to the `loadView` method in the `view-base.ts`)?

**Comment by vakrilov** [comment]
closing in favor of @6255
